### PR TITLE
LSM/Compaction: Pacing review comments/assertions/refactors

### PR DESCRIPTION
**Pull request by sentientwaffle**
This is a bunch of miscellaneous:
- documentation "improvements" (hopefully)
- assertions
- small/local refactors

as a follow-up "review" for https://github.com/tigerbeetle/tigerbeetle/pull/1660.

### CODE REVIEW

File: src/lsm/forest.zig [hunk 1]
```diff
@@ -1344,14 +1357,10 @@ fn CompactionPipelineType(comptime Forest: type, comptime Grid: type) type {
...
-                i -= 1;
-
-                // We need to run this for all compactions that ran acquire - even if they
-                // transitioned to being finished, so we can't just use bar_active.
-                if (!self.beat_reserved.isSet(i)) continue;
-
+            // We need to run this for all compactions that ran acquire - even if they
+            // transitioned to being finished, so we can't just use bar_active.
+            var compactions_reserved = self.beat_reserved.iterator(.{});
+            while (compactions_reserved.next()) |i| {
```
Comment (sentientwaffle) [inline src/lsm/forest.zig:1364]: I didn't see any reason why we should forfeit in reverse order, but lmk if I've missed something.

File: src/lsm/forest.zig [hunk 2]
```diff
@@ -380,7 +380,7 @@ pub fn ForestType(comptime _Storage: type, comptime groove_cfg: anytype) type {
                 @divExact(constants.lsm_batch_multiple, 2) - 1;
             const half_beat = compaction_beat == @divExact(constants.lsm_batch_multiple, 2);
             const last_beat = compaction_beat == constants.lsm_batch_multiple - 1;
-            assert(@intFromBool(first_beat) + @intFromBool(last_half_beat) +
+            assert(@as(usize, @intFromBool(first_beat)) + @intFromBool(last_half_beat) +
                 @intFromBool(half_beat) + @intFromBool(last_beat) <= 1);
```
Comment (sentientwaffle) [inline src/lsm/forest.zig:384]: `@intFromBool + @intFromBool + ...` will integer-overflow if more than one of the booleans are true.

File: src/lsm/compaction.zig [hunk 3]
```diff
@@ -965,19 +981,16 @@ pub fn CompactionType(
                 const value_block_checksums = index_schema.data_checksums_used(index_block);
 
                 var free_blocks_used: usize = 0;
+                // Read at most half of the blocks, since the other half are reserved for the
```
Comment (cb22) [inline src/lsm/compaction.zig]: This isn't the case here; all of `source_value_blocks[...]` are reserved for input blocks, but we want to smooth out the reading over multiple steps to reduce latency spikes.

File: src/lsm/forest.zig (hunk 2 above)
Comment (cb22) [reply src/lsm/forest.zig:384]: "This is not the panic you're looking for..."

File: src/lsm/forest.zig (hunk 1 above)
Comment (cb22) [reply src/lsm/forest.zig:1364]: Yeah - it's not needed!

File: src/lsm/compaction.zig (hunk 3 above)
Comment (sentientwaffle) [reply src/lsm/compaction.zig]: Updated to

```
                // Read at most half of the blocks, to smooth out reading over multiple steps.
```

**Review by cb22 — APPROVED** [review]

**Pull request merged**